src/systems/particle_emitter/ParticleEmitter.cc

```diff
@@ -0,0 +1,191 @@
+
+  // Size.
+  ignition::math::Vector3d size = ignition::math::Vector3d::One;
+  if (_sdf->HasElement("size"))
+    size = _sdf->Get<ignition::math::Vector3d>("size");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_size(), size);
+
+  // Rate.
+  this->dataPtr->emitter.set_rate(_sdf->Get<double>("rate", 10).first);
+
+  // Duration.
+  this->dataPtr->emitter.set_duration(_sdf->Get<double>("duration", 0).first);
+
+  // Emitting.
+  this->dataPtr->emitter.set_emitting(_sdf->Get<bool>("emitting", false).first);
+
+  // Particle size.
+  size = ignition::math::Vector3d::One;
+  if (_sdf->HasElement("particle_size"))
+    size = _sdf->Get<ignition::math::Vector3d>("particle_size");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_particle_size(), size);
+
+  // Lifetime.
+  this->dataPtr->emitter.set_lifetime(_sdf->Get<double>("lifetime", 5).first);
+
+  // Material.
+  if (_sdf->HasElement("material"))
+  {
+    auto materialElem = _sdf->GetElementImpl("material");
+    sdf::Material material;
+    material.Load(materialElem);
+    ignition::msgs::Material materialMsg = convert<msgs::Material>(material);
+  }
+
+  // Min velocity.
+  this->dataPtr->emitter.set_min_velocity(
+    _sdf->Get<double>("min_velocity", 1).first);
+
+  // Max velocity.
+  this->dataPtr->emitter.set_max_velocity(
+    _sdf->Get<double>("max_velocity", 1).first);
+
+  // Color start.
+  ignition::math::Color color = ignition::math::Color::White;
+  if (_sdf->HasElement("color_start"))
+    color = _sdf->Get<ignition::math::Color>("color_start");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_color_start(), color);
+
+  // Color end.
+  color = ignition::math::Color::White;
+  if (_sdf->HasElement("color_end"))
+    color = _sdf->Get<ignition::math::Color>("color_end");
+  ignition::msgs::Set(this->dataPtr->emitter.mutable_color_end(), color);
+
+  // Scale rate.
+  this->dataPtr->emitter.set_scale_rate(
+    _sdf->Get<double>("scale_rate", 1).first);
+
+  // Color range image.
+  this->dataPtr->emitter.set_color_range_image(
+    _sdf->Get<std::string>("color_range_image", "").first);
+
+  igndbg << "Loading particle emitter:" << std::endl
+         << this->dataPtr->emitter.DebugString() << std::endl;
+
+  // Create an audio source entity.
+  auto entity = _ecm.CreateEntity();
+  if (entity == kNullEntity)
+  {
+    ignerr << "Failed to create a particle emitter entity" << std::endl;
+    return;
+  }
+
+  // Create components.
+  _ecm.CreateComponent(entity,
+    components::Name("particle_emitter_" + this->dataPtr->emitter.name()));
+
+  particles::Emitter emitterData;
+  emitterData.data.CopyFrom(this->dataPtr->emitter);
+  _ecm.CreateComponent(entity, components::ParticleEmitter(emitterData));
+
+  _ecm.CreateComponent(entity, components::Pose(pose));
+}
+
+//////////////////////////////////////////////////
+void ParticleEmitter::PreUpdate(const ignition::gazebo::UpdateInfo &/*_info*/,
```

Comment:
I guess the implementation of this function will come in a future PR, right?

Reply:
For the basic functionality I don't need this function, as the component is propagated into the ecm, and then, into the rendering thread, where the particle emitter is instantiated.